Config refactor + add option for disabling logging on world loading

dxnnv opened the pull request:
This PR cleans up the configuration file, and adds the option to disable hologram loading log messages when worlds are loaded. The new config.yml is as follows:

```yaml
saving:
  autosave:
    enabled: true # Whether autosave is enabled.
    interval: 15 # The interval at which autosave is performed in minutes.
  save_on_changed: true # Whether the plugin should save holograms when they are changed.
logging:
  log_level: INFO # The log level for the plugin (DEBUG, INFO, WARN, ERROR).
  log_on_world_load: true # Whether hologram loading should be logged on world loading. Disable this if you load worlds dynamically to prevent console spam.
  version_notification: true # Whether the plugin should send notifications for new updates.
visibility_distance: 20 # The default visibility distance for holograms.
register_commands: true # Whether the plugin should register its commands.
```

This simply organizes the configuration for improved readability by nesting common options and ordering them properly. `mute_version_notifications` has been refactored to `logging.version_notifications` to make the option more intuitive (true = sends notifications, false = does not send notifications). All files and method names have been updated accordingly.

The option `logging.log_on_world_load` provides the ability to disable messages such as:
`[FancyHolograms] (FancyHolograms-Holograms) INFO: Loading holograms for <worldname>`
This allows server operators to prevent FancyHolograms from spamming console on servers that load/unload worlds dynamically (especially worlds that have no holograms, which makes these messages redundant).

dxnnv committed:
Organize config options, add hologramLoadLogging, invert version notifications

dxnnv committed:
Organize config options, add hologramLoadLogging, invert version notifications

dxnnv committed:
Enable version notifications by default

dxnnv committed:
Enable version notifications by default

dxnnv committed:
Only log hologram loading on world load if hologramLoadLogging is true

dxnnv committed:
Only log hologram loading on world load if hologramLoadLogging is true

dxnnv committed:
Add config updater, define keys via constants

dxnnv committed:
Add config updater, define keys via constants

dxnnv committed:
Refactor to make this less messy

dxnnv committed:
Refactor to make this less messy

dependabot[bot] committed:
chore(deps): bump de.oliver:FancyLib from 35 to 36 (#202)

Bumps de.oliver:FancyLib from 35 to 36.

---
updated-dependencies:
- dependency-name: de.oliver:FancyLib
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

OliverSchlueter committed:
Update build.yml

OliverSchlueter committed:
Add update visibility interval configuration for holograms

OliverSchlueter committed:
Change hologram visibility update interval type from long to int

dependabot[bot] committed:
chore(deps): bump de.oliver:FancyNpcs from 2.4.2 to 2.4.4 (#204)

Bumps de.oliver:FancyNpcs from 2.4.2 to 2.4.4.

---
updated-dependencies:
- dependency-name: de.oliver:FancyNpcs
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump org.lushplugins:ChatColorHandler from 5.1.2 to 5.1.3 (#198)

Bumps org.lushplugins:ChatColorHandler from 5.1.2 to 5.1.3.

---
updated-dependencies:
- dependency-name: org.lushplugins:ChatColorHandler
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

OliverSchlueter commented on src/main/java/de/oliver/fancyholograms/FancyHologramsConfiguration.java:
```diff
@@ -3,97 +3,170 @@
     private boolean registerCommands;
+    private static final String CONFIG_REGISTER_COMMANDS = "register_commands";
+
+    private static final String CONFIG_REPORT_ERRORS_TO_SENTRY = "logging.report_errors_to_sentry";
```
This should be only `report_errors_to_sentry`. We want to remove that key and the logging section didn't exist before

OliverSchlueter commented:
Otherwise LGTM. Could you also resolve the merge conflicts?

dependabot[bot] committed:
chore(deps): bump com.viaversion:viaversion-api from 5.2.1 to 5.3.0 (#209)

Bumps com.viaversion:viaversion-api from 5.2.1 to 5.3.0.

---
updated-dependencies:
- dependency-name: com.viaversion:viaversion-api
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump io.papermc.paper:paper-api (#208)

Bumps io.papermc.paper:paper-api from 1.21.4-R0.1-SNAPSHOT to 1.21.5-R0.1-SNAPSHOT.

---
updated-dependencies:
- dependency-name: io.papermc.paper:paper-api
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump io.papermc.hangar-publish-plugin from 0.1.2 to 0.1.3 (#206)

Bumps io.papermc.hangar-publish-plugin from 0.1.2 to 0.1.3.

---
updated-dependencies:
- dependency-name: io.papermc.hangar-publish-plugin
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dxnnv committed:
fix: optimize imports

dxnnv committed:
fix: remove redundant cast

dxnnv committed:
refactor: extract method

dxnnv committed:
refactor: declare type explicitly

dxnnv committed:
refactor: utilize StringBuilder

dxnnv committed:
refactor: explicitly declare long values

dxnnv committed:
refactor: add annotations and optimize imports

dxnnv committed:
style: remove redundant semicolon

dxnnv committed:
feat: include update visibility interval in config

dxnnv committed:
refactor: utilize switch statement

dxnnv committed:
Merge remote-tracking branch 'origin/main'

# Conflicts:
#	api/src/main/java/de/oliver/fancyholograms/api/HologramConfiguration.java
#	src/main/java/de/oliver/fancyholograms/FancyHologramsConfiguration.java

dxnnv committed:
Forgot some update visibility interval stuff

dxnnv commented:
Should be good to merge now, combined changes to resolve conflicts. Feel free to lmk if anything else is needed!

OliverSchlueter commented on src/main/java/de/oliver/fancyholograms/commands/FancyHologramsCMD.java:
```diff
@@ -105,7 +98,7 @@ public boolean execute(@NotNull CommandSender sender, @NotNull String label, @No
 
     @Override
-    public @NotNull List<String> tabComplete(@NotNull CommandSender sender, @NotNull String label, @NotNull String[] args) throws IllegalArgumentException {
+    public @NotNull List<String> tabComplete(@NotNull CommandSender sender, @NotNull String label, String @NotNull [] args) throws IllegalArgumentException {
```
I would prefer String[]

OliverSchlueter commented on src/main/java/de/oliver/fancyholograms/commands/FancyHologramsCMD.java:
```diff
@@ -29,7 +26,7 @@ public FancyHologramsCMD(@NotNull final FancyHolograms plugin) {
 
     @Override
-    public boolean execute(@NotNull CommandSender sender, @NotNull String label, @NotNull String[] args) {
+    public boolean execute(@NotNull CommandSender sender, @NotNull String label, String @NotNull [] args) {
```
Doesn't look correct. I would prefer `@NotNull String[]`

dxnnv committed:
style: move annotations

OliverSchlueter reviewed (approved):
Thank you!
